bootloader/src/main.c

```diff
@@ -414,8 +417,19 @@ int main(void)
             // Initialize user application's Stack Pointer
             __set_MSP(*(__IO uint32_t*) ApplicationAddress);
 
-            // Set IWDG Timeout to 5 secs based on platform specific system flags
-            IWDG_Reset_Enable(5 * TIMING_IWDG_RELOAD);
+            uint8_t disable_iwdg = 0;
+#ifdef CHECK_FIRMWARE
+            // Pre-0.7.0 firmwares were expecting IWDG flag to be set in the DCT, now it's stored in
+            // the backup registers. As a workaround, we disable IWDG if an older firmware is detected
+            const uint16_t module_ver = get_main_module_version();
+            if (module_ver >= 0 && module_ver < SYSTEM_MODULE_VERSION_0_7_0_RC1) {
```

Comment:
It's the opposite: we don't want to enable IWDG for module versions < 200 (0.7.0-rc.1), since older firmwares won't be able detect if IWDG is running. In other words, we're enabling IWDG for any firmware version in the range 200-65535, which includes most of the potentially corrupted values and erased state.